drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -2709,7 +2716,9 @@ static void rswitch_fib_event_add(struct rswitch_fib_event_work *fib_work)
 	new_routing_list->dev = dev;
 	INIT_LIST_HEAD(&new_routing_list->param_list);
 
+	mutex_lock(&dev->priv->ipv4_forward_lock);
```

Comment:
It seems like this lock can be redundant because forward flag status has to be kept in mind only when you setup hw.